### ISSUE TITLE
Is it possible to also update a Chart.yaml for a Helm Chart?

### ISSUE THREAD

**Issue by jseiser** [python-semantic-release/python-semantic-release#601]
We use this software, to create changelogs, and tag versions in gitlab.  Is it possible to also update the version, in a Chart.yaml for our Helm Charts?

```
❯ tree -L 2
.
├── CHANGELOG.md
├── console
│   ├── charts
│   ├── Chart.yaml
│   ├── README.md
│   ├── templates
│   └── values.yaml
├── Pipfile
├── Pipfile.lock
├── pyproject.toml
└── README.md
```

```
❯ cat console/Chart.yaml 
apiVersion: v2
name: console
description: A Helm chart for Kubernetes

# A chart can be either an 'application' or a 'library' chart.
#
# Application charts are a collection of templates that can be packaged into versioned archives
# to be deployed.
#
# Library charts provide useful utilities or functions for the chart developer. They're included as
# a dependency of application charts to inject those utilities and functions into the rendering
# pipeline. Library charts do not define any templates and therefore cannot be deployed.
type: application

# This is the chart version. This version number should be incremented each time you make changes
# to the chart and its templates, including the app version.
# Versions are expected to follow Semantic Versioning (https://semver.org/)
version: 0.1.0

# This is the version number of the application being deployed. This version number should be
# incremented each time you make changes to the application. Versions are not expected to
# follow Semantic Versioning. They should reflect the version the application is using.
# It is recommended to use it with quotes.
appVersion: "v1.11.0"
```

```
❯ cat pyproject.toml 
[build-system]
requires = ["setuptools"]

[project]
name = "console"
version = "0.1.12"

[tool.semantic_release]
version_toml = ["version_toml = ["pyproject.toml:project.version"]"]
branch = "develop"
build_command = false
upload_to_repository = false
hvcs = "gitlab"
hvcs_domain = "gitlab.company.network"
parser_angular_patch_types = "fix,perf,refactor,build,style"
changelog_sections="feature,fix,breaking,documentation,performance,refactor,build,style,ci,Other"

[tool.setuptools]
packages = []
```

Since `Chart.yaml` is not a toml file, I assume this just isnt possible?

**Comment by bernardcooke53** [comment]
Hey @jseiser - it should absolutely be possible, I suggest you use [version_variable](https://python-semantic-release.readthedocs.io/en/latest/configuration.html#version-variable) as follows:

```toml
[tool.semantic_release]
version_toml = ["pyproject.toml:project.version"]  # for the package metadata
version_variable = [
    "console/Chart.yaml:^version: {version}",  # for the chart version
    "console/Chart.yaml:^appVersion: \"{version}\"",  # for the app version
]
```

**Comment by jseiser** [comment]
@bernardcooke53 

```
debug: Writing new version number: path=PosixPath('a3s-high-adapter/Chart.yaml') pattern='^version: {version} *[:=] *["\\\'](\\d+\\.\\d+\\.\\d+(-beta\\.\\d+)?)["\\\']' num_matches=0
```


```
[project]
name = "a3s-high-adapter"
version = "0.1.3"

[tool.semantic_release]
version_toml = ["pyproject.toml:project.version"]
version_variable = [
    "a3s-high-adapter/Chart.yaml:^version: {version}"
]
```

```
❯ bat a3s-high-adapter/Chart.yaml 
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: a3s-high-adapter/Chart.yaml
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ apiVersion: v2
   2   │ name: a3s-high-adapter
   3   │ description: A Helm chart for Kubernetes
   4   │ 
   5   │ # A chart can be either an 'application' or a 'library' chart.
   6   │ #
   7   │ # Application charts are a collection of templates that can be packaged into versioned archives
   8   │ # to be deployed.
   9   │ #
  10   │ # Library charts provide useful utilities or functions for the chart developer. They're included as
  11   │ # a dependency of application charts to inject those utilities and functions into the rendering
  12   │ # pipeline. Library charts do not define any templates and therefore cannot be deployed.
  13   │ type: application
  14   │ 
  15   │ # This is the chart version. This version number should be incremented each time you make changes
  16   │ # to the chart and its templates, including the app version.
  17   │ # Versions are expected to follow Semantic Versioning (https://semver.org/)
  18   │ version: 0.1.3
```

**Comment by bernardcooke53** [comment]
@jseiser I think it's failing to match because your version isn't quoted - the simplest fix might be just to change your Chart.yaml to have:

```yaml
version: "0.1.3"
```

I know it's valid YAML without quotes, but it'll be interpreted the same by helm.

**Comment by jseiser** [comment]
@bernardcooke53 

No Change.

```
❯ bat -p tinysoc/Chart.yaml 
apiVersion: v2
name: tinysoc
description: A Helm chart for Kubernetes

# A chart can be either an 'application' or a 'library' chart.
#
# Application charts are a collection of templates that can be packaged into versioned archives
# to be deployed.
#
# Library charts provide useful utilities or functions for the chart developer. They're included as
# a dependency of application charts to inject those utilities and functions into the rendering
# pipeline. Library charts do not define any templates and therefore cannot be deployed.
type: application

# This is the chart version. This version number should be incremented each time you make changes
# to the chart and its templates, including the app version.
# Versions are expected to follow Semantic Versioning (https://semver.org/)
version: "0.0.0"
```

```
❯ semantic-release --noop version --verbosity=DEBUG
Creating new version
debug: get_current_version_by_config_file()
debug: Parsing current version: path=PosixPath('tinysoc/Chart.yaml') pattern='^version: {version} *[:=] *["\\\'](\\d+\\.\\d+\\.\\d+(-beta\\.\\d+)?)["\\\']' num_matches=0
debug: Regex matched version: 0.0.0
debug: get_current_version_by_config_file -> 0.0.0
debug: get_current_release_version_by_commits()
debug: Checking commit e4d2aab72871191d5305f4afacf19badcbf6f715
debug: Checking commit a1ea544f12bf70d6b36a3ff0fc9111114c88d415
debug: get_current_release_version_by_commits -> 0.0.0
Current version: 0.0.0, Current release version: 0.0.0
debug: evaluate_version_bump('0.0.0', None)
debug: Reference v0.0.0 does not exist, considering entire history
debug: parse_commit_message('feat(release): initial release')
debug: parse_commit_message -> ParsedCommit(bump=2, type='feature', scope='release', descriptions=['initial release'], breaking_descriptions=[])
debug: parse_commit_message('Initial commit')
debug: Ignoring UnknownCommitMessageStyleError: Unable to parse the given commit message: Initial commit
debug: Commits found since last release: 2
debug: evaluate_version_bump -> minor
debug: get_new_version(0.0.0, '0.0.0', 'minor', False, True)
debug: get_new_version -> 0.1.0
warning: No operation mode. Should have bumped from 0.0.0 to 0.1.0
```

**Comment by bernardcooke53** [comment]
I think you're hitting the same root cause as https://github.com/python-semantic-release/python-semantic-release/issues/567 now - if you're tagging with version numbers, you can try using `version_source = "tag"`, otherwise you probably need to create a commit with the version number as a message in order to introduce PSR to the project - e.g.

```bash
git commit --allow-empty -m "0.1.3"
```

**Comment by aaronborden-rivian** [comment]
I can confirm that this configuration works for me.

```yaml
# Chart.yaml
version: "0.1.0"

# pyproject.toml
[tool.semantic_release]
version_variable = [
    "pyproject.toml:version",
    "Chart.yaml:version"
]
```

**Comment by codejedi365** [comment]
@aaronborden-rivian, thank you for confirming success for even the most recent version.

**Comment by codejedi365** [comment]
### :tada: This resolution has been included in version 9.8.9 :tada:

The release is available on:
- [PyPi package](https://pypi.org/project/python-semantic-release/9.8.9/)
- [GitHub release](https://github.com/python-semantic-release/python-semantic-release/releases/tag/v9.8.9)

----

 @jseiser / @aaronborden-rivian , I know we resolved this question earlier but I wanted to let you know of the latest release which has a more flexible regular expression now that you will no longer need quotes around your version inside your yaml file.
